akka-remote/src/main/protobuf/ArteryControlFormats.proto

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+option java_package = "akka.remote";
+option optimize_for = SPEED;
+
+message Quarantined {
+  required UniqueAddress from = 1;
+  required UniqueAddress to = 2;
+}
+
+// Generic message declaration that is used for all types of message that (currently) have a single
+// address field. A message that needs to changed later can be cloned from this one and then adapted.
+// ActorSystemTerminating
+// ActorSystemTerminating.Ack
+// OutboundHandshake.HandshakeReq
+// OutboundHandshake.HandshakeRsp
+message MessageWithAddress {
+  required UniqueAddress address = 1;
+}
+
+// CompressionProtocol.ActorRefCompressionAdvertisement
+// CompressionProtocol.ClassManifestCompressionAdvertisement
+message CompressionTableAdvertisement {
+  required UniqueAddress from = 1;
+  required uint32 tableVersion = 2;
+
+  // actual Map is represented by separate sequences of keys and values,
+  // relies on both sequences using the same order so that corresponding entries can be
+  // associated again when deserializing
+  repeated string keys = 3;
+  repeated uint32 values = 4;
+}
+
+// CompressionProtocol.ActorRefCompressionAdvertisementAck
+// CompressionProtocol.ClassManifestCompressionAdvertisementAck
+message CompressionTableAdvertisementAck {
+  required UniqueAddress from = 1;
+  required uint32 version = 2;
+}
+
+// SystemMessageDelivery.SystemMessageEnvelope
+message SystemMessageEnvelope {
+  required bytes message = 1; // FIXME: is that the right way? WireFormats.SerializedMessage underneath
```

Comment:
I mean that `SerializedMessage` is two levels of wrapping, while the `SelectionEnvelope` approach is only one level